src/cosalib/container_manifest.py

```diff
@@ -20,31 +20,35 @@ def create_local_container_manifest(repo, tag, images) -> dict:
     return json.loads(manifest_info)
 
 
-def local_container_manifest_exists(repo, tag):
+def local_container_manifest_or_image_exists(repo, tag):
     '''
-    Delete local manifest list
+    Delete local manifest list or image associated with repo:tag
     @param repo str registry repository
-    @param tag str manifest tag
+    @param tag str tag
     '''
-    cmd = ["podman", "manifest", "exists", f"{repo}:{tag}"]
-    cp = runcmd(cmd, check=False)
-    # The commands returns 0 (exists), 1 (doesn't exist), 125 (other error)
-    if cp.returncode == 125:
-        if cp.stdout:
-            print(f" STDOUT: {cp.stdout.decode()}")
-        if cp.stderr:
-            print(f" STDERR: {cp.stderr.decode()}")
-        raise Exception("Error encountered when checking if manifest exists")
-    return cp.returncode == 0
+    cmds = [["podman", "manifest", "exists", f"{repo}:{tag}"],
+            ["podman", "image", "exists", f"{repo}:{tag}"]]
+    for cmd in cmds:
+        cp = runcmd(cmd, check=False)
+        # The commands returns 0 (exists), 1 (doesn't exist), 125 (other error)
+        if cp.returncode == 125:
+            if cp.stdout:
+                print(f" STDOUT: {cp.stdout.decode()}")
+            if cp.stderr:
+                print(f" STDERR: {cp.stderr.decode()}")
+            raise Exception("Error encountered when checking if manifest exists")
+        if cp.returncode == 0:
+            return True
+    return False
 
 
-def delete_local_container_manifest(repo, tag):
+def delete_local_container_imgref(repo, tag):
     '''
-    Delete local manifest list
+    Delete local manifest list or image associated with repo:tag
     @param repo str registry repository
     @param tag str manifest tag
     '''
-    cmd = ["podman", "manifest", "rm", f"{repo}:{tag}"]
+    cmd = ["podman", "image", "rm", f"{repo}:{tag}"]
```

Comment:
I'm not sure this will still work in the manifest list case. Should there be a list of commands like in `local_container_manifest_or_image_exists`?

Reply:
I tested it and it appears to work for both a non manifest listed reference and a manifest listed reference.

Reply:
SGTM, thanks for the info!